File: java/eyes.sdk.core/src/main/java/com/applitools/eyes/options/LayoutBreakpointsOptions.java
package com.applitools.eyes.options;

import com.applitools.utils.ArgumentGuard;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class LayoutBreakpointsOptions {

    /**
     * use default layout breakpoints
     */
    private Boolean isLayoutBreakpoints;

    /**
     * explicit list of layout breakpoints
     */
    private final List<Integer> layoutBreakpoints = new ArrayList<>();

    /**
     * reload after layout breakpoints
     */
    private Boolean reload;

    public LayoutBreakpointsOptions breakpoints() {
        isLayoutBreakpoints = true;
        layoutBreakpoints.clear();
        return this;
    }

    public LayoutBreakpointsOptions breakpoints(Boolean shouldSet) {
        isLayoutBreakpoints = shouldSet;
        layoutBreakpoints.clear();
        return this;
    }

    public LayoutBreakpointsOptions breakpoints(int[] breakpoints) {
        return breakpoints(
                Arrays.stream(breakpoints).boxed().toArray(Integer[]::new)
        );
    }

    public LayoutBreakpointsOptions breakpoints(Integer... breakpoints) {
        isLayoutBreakpoints = null;
        layoutBreakpoints.clear();

        if (breakpoints == null || breakpoints.length == 0) {
            return this;
        }

        for (int breakpoint : breakpoints) {
            ArgumentGuard.greaterThanZero(breakpoint, "breakpoint");
            layoutBreakpoints.add(breakpoint);
        }

        Collections.sort(layoutBreakpoints);
        return this;
    }

    public LayoutBreakpointsOptions reload() {
        reload = true;
        return this;
    }
    public LayoutBreakpointsOptions reload(Boolean shouldReload) {
        reload = shouldReload;
        return this;
    }

    public Boolean getReload() {
        return reload;
    }

    public Boolean isLayoutBreakpoints() { return isLayoutBreakpoints; }

    public List<Integer> getLayoutBreakpoints() { return layoutBreakpoints; }

}

File: dotnet/Eyes.Images.DotNet/Options/LayoutBreakpointsOptions.cs
using System.Collections.Generic;
using Applitools.Utils;
using Newtonsoft.Json;

namespace Applitools
{
    [JsonConverter(typeof(LayoutBreakpointsConverter))]
    public class LayoutBreakpointsOptions
    {
        public LayoutBreakpointsOptions()
        {
        }

        internal LayoutBreakpointsOptions(LayoutBreakpointsOptions options)
        {
            if (options == null)
            {
                return;
            }
            foreach (var bp in options.layoutBreakpoints_)
            {
                layoutBreakpoints_.Add(bp);
            }

            isLayoutBreakpoints_ = options.isLayoutBreakpoints_;
            reload_ = options.reload_;
        }

        private bool? isLayoutBreakpoints_;

        /// <summary>
        /// explicit list of layout breakpoints
        /// </summary>
        [JsonProperty("breakpoints")]
        private readonly List<int> layoutBreakpoints_ = new List<int>();

        /// <summary>
        /// reload after layout breakpoints
        /// </summary>
        [JsonProperty("reload")]
        private bool? reload_;

        public LayoutBreakpointsOptions Breakpoints()
        {
            isLayoutBreakpoints_ = true;
            layoutBreakpoints_.Clear();
            return this;
        }

        public LayoutBreakpointsOptions Breakpoints(bool? shouldSet)
        {
            isLayoutBreakpoints_ = shouldSet;
            layoutBreakpoints_.Clear();
            return this;
        }

        public LayoutBreakpointsOptions Breakpoints(params int[] breakpoints)
        {
            isLayoutBreakpoints_ = null;
            layoutBreakpoints_.Clear();

            if (breakpoints == null || breakpoints.Length == 0)
            {
                return this;
            }

            foreach (int breakpoint in breakpoints)
            {
                ArgumentGuard.GreaterThan(breakpoint, 0, nameof(breakpoint));
                layoutBreakpoints_.Add(breakpoint);
            }

            layoutBreakpoints_.Sort();
            return this;
        }

        public LayoutBreakpointsOptions Reload()
        {
            reload_ = true;
            return this;
        }

        public LayoutBreakpointsOptions Reload(bool shouldReload)
        {
            reload_ = shouldReload;
            return this;
        }

        public bool? GetReload()
        {
            return reload_;
        }

        public bool? IsLayoutBreakpoints()
        {
            return isLayoutBreakpoints_;
        }

        public List<int> GetLayoutBreakpoints()
        {
            return layoutBreakpoints_;
        }
    }
}